MouseAimFlight/MouseAimVesselModule.cs

```diff
@@ -173,49 +175,49 @@ void MouseAimPilot(FlightCtrlState s)
-                pitchYawOverrideMouseAim = true;
+                FlyToPosition(s, vesselTransform.up * 5000f + vessel.CoM);
                 return;
             }
             else
-                pitchYawOverrideMouseAim = false;
-
-            upDirection = VectorUtils.GetUpDirection(vesselTransform.position);
-
-            FlyToPosition(s, targetPosition + vessel.CoM);
-        }
-
-        void UpdateMouseCursorForCameraRotation()
-        {
-            if (pitchYawOverrideMouseAim)
             {
-                targetPosition = vesselTransform.up * 5000f;
+                FlyToPosition(s, targetPosition + vessel.CoM);
             }
-            else
-            {
-                Vector3 mouseDelta;
+        }
 
-                if (freeLook)
-                    mouseDelta = Vector3.zero;
-                else
-                    mouseDelta = new Vector3(Input.GetAxisRaw("Mouse X"), Input.GetAxisRaw("Mouse Y")) * MouseAimSettings.MouseSensitivity;
+        void UpdateMouseCursorAndCameraRotation()
+        {
+            Vector3 mouseDelta;
 
-                if (MouseAimSettings.InvertXAxis)
-                    mouseDelta.x *= -1;
-                if (MouseAimSettings.InvertYAxis)
-                    mouseDelta.y *= -1;
+            if (freeLook)
+                mouseDelta = Vector3.zero;
+            else
+                mouseDelta = new Vector3(Input.GetAxisRaw("Mouse X"), Input.GetAxisRaw("Mouse Y")) * MouseAimSettings.MouseSensitivity;
 
-                Transform cameraTransform = FlightCamera.fetch.mainCamera.transform;
+            if (MouseAimSettings.InvertXAxis)
+                mouseDelta.x *= -1;
+            if (MouseAimSettings.InvertYAxis)
+                mouseDelta.y *= -1;
 
-                Vector3d localTarget = cameraTransform.InverseTransformDirection(targetPosition);
-                localTarget += mouseDelta;
-                localTarget.Normalize();
-                localTarget *= 5000f;
+            Transform cameraTransform = FlightCamera.fetch.mainCamera.transform;
 
-                targetPosition = cameraTransform.TransformDirection(localTarget);
+            Vector3d localTarget = cameraTransform.InverseTransformDirection(targetPosition);
+            localTarget += mouseDelta;
+            localTarget.Normalize();
+            if (!freeLook)
+            {
+                float targetPitch = Clamp(FlightCamera.CamPitch - Mathf.Asin((float)localTarget.y) + 0.15f, -Mathf.PI * 0.47f, Mathf.PI * 0.47f);
+                float targetHdg = FlightCamera.CamHdg + (float)Math.Atan2(localTarget.x, localTarget.z);
+                FlightCamera.CamPitch = Mathf.Lerp(FlightCamera.CamPitch, targetPitch, 1 - Mathf.Exp(-7.5f * Time.fixedDeltaTime));
+                FlightCamera.CamPitch = Clamp(FlightCamera.CamPitch, -Mathf.PI * 0.47f, Mathf.PI * 0.47f);
+                FlightCamera.CamHdg = Mathf.Lerp(FlightCamera.CamHdg, targetHdg, 1 - Mathf.Exp(-7.5f * Time.fixedDeltaTime)); //Frame-independent update
```

Comment:
I have absolutely no idea what is going on here :P

Reply:
ehhh, what I did was setting the Camera's Hdg and Pitch towards whre localtarget is pointed at...
I've noticed this could be replaced by setting targetDirection directly. Gonna delete these redundant codes.